### ISSUE TITLE
Status of this library?

### ISSUE THREAD

**Issue by retepz** [wildbit/mailmason#76]
Not exactly sure who or where to reach out, but is this library still actively maintained? I know postmark still has a link to this repo but my biggest concern with using this library are the outdated dependencies.  A lot of them haven't been updated in years.

Related, are there plans to using something a little more modern like webpack instead of grunt?  I'm looking into expanding some of the functionality and would prefer to use es modules instead of the old commonJS way of doing things. 

Thanks

**Comment by DWboutin** [comment]
@pzimmermaninmo Hi! I don't think this project is actively maintained, but it worked well for us. I built a complete mail builder less than 3 months ago and it works well.

If you are using Postmark, it's the best solution I found.

**Comment by retepz** [comment]
> @pzimmermaninmo Hi! I don't think this project is actively maintained, but it worked well for us. I built a complete mail builder less than 3 months ago and it works well.
> 
> If you are using Postmark, it's the best solution I found.

Yeah we actually started using it and it seems to work its just unfortunate its no longer being maintained.  If it breaks one day we might be SOL.  (Not to mention all the security vulnerabilities with these outdated libraries.  Hell, one of the dependencies spits out this gem:
```
warning time-grunt@2.0.0: Deprecated because Grunt is practically unmaintained. Move on to something better. This package will continue to work with Grunt v1, but it will not receive any updates.
```

**Comment by dprensha** [comment]
> Yeah we actually started using it and it seems to work its just unfortunate its no longer being maintained. If it breaks one day we might be SOL. 

We are in the same boat. I think it's more of the grunt "helper" dependencies that aren't being maintained. At this point we are debating rolling our own email build process so that we can mitigate this risk and also reduce the vulnerabilities reported.

Currently I am getting: 37 vulnerabilities (4 low, 6 moderate, 22 high, 5 critical). We customized this solution a bit so it may not match exactly what everyone else would see

**Comment by iampava** [comment]
I think using React Email and the Postmark Node SDK we could achieve the same thing.

Anybody tried it? I was just playing with it locally and seems promising.

**Comment by melv-n** [comment]
Ran into the same challenge. Seeing MailMason being pushed/advertised on the storefront and inside the product gives the impression it's at least maintained on a vulnerability level. Sad to see it's not maintained.

**Comment by dvins** [comment]
@iampava @melv-n MailMason was beautiful, but Hands down, React Email + PostMark is now the way to go.

**Comment by melv-n** [comment]
> @iampava @melv-n MailMason was beautiful, but Hands down, React Email + PostMark is now the way to go.

Thanks for the reply, I actually found about React Email moments after I posted here and it sure looks promising!

Do you render + upload the templates to Postmark (with template variables for dynamic content), or do you use a Node server with Postmark js sdk to render + send the actual email?

**Comment by dvins** [comment]
@melv-n We deliver dozens of notification types across multiple channels: In-app, SMS/TXT, and email. Complicating this, we're a multisided SaaS platform. A considerable volume of notifications need to be branded per customer and even further customized per our customer's customer. 

In terms of how we do this, we're primarily a TypeScript + NestJs shop on the backend, and heavily leverage an event-driven messaging architecture based around CloudEvents and Redis backed queues with observability and reporting via Postgres.

From a workflow perspective, we have a SendNotification* command which is picked up by a handler. It scavenges the relevant entity data and fetches the notification templates and branding configuration. It then renders the output for each delivery channel and user. The rendered output, especially HTML email, can be considerably large, so we use a [claim check pattern](https://www.enterpriseintegrationpatterns.com/patterns/messaging/StoreInLibrary.html) to offload the rendered bits to cloud storage as opposed to publishing a massive NotificationRendered event. This also lets us address other compliance needs.

A subsequent handler picks up the notification rendered event, fetches the bits, and then attempts delivery via the appropriate channels and mechanisms; interfacing with third-party services like Postmark along the way.

This too poses challenges. We not only send transactional emails to our platform users, but we also have to send emails to our the customer's customers using their branded domains. Topping it all off, we have a class of users who enable email integration such that a message (to their customers) is sent directly from their Microsoft Email or Google Email.
